client/isolated.go

```diff
@@ -0,0 +1,124 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/notary"
+	"github.com/docker/notary/cryptoservice"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+	"io"
+	"io/ioutil"
+)
+
+// AddTargetToFile adds the target to the file found at filePath, and writes the updated and re-signed version
+// to the outBuf.
+// If signingKeys are provided, they are used to sign the file. If lookupKeys are provided, they
+// will be looked up in the CryptoService and used for signing where found.
+// If _both_ signingKeys, and lookupKeys are provided, the public keys in lookupKeys will be added to signingKeys
+// and the merged list will be used for signing.
+// If _neither_ signingKeys, no lookupKeys are provided, the key IDs associated with any existing signatures are
```

Comment:
the function internally looks up keys based on signatures if nothing else was provided. I felt that was better than having to pre-parse the file and do the lookup of keys based on signatures. I figured in some cases one might only have the public key (for example, in the case the private key is in a Yubikey), so `lookupKeys` is basically the list of public keys provided on the CLI.